### PR TITLE
Add feature for cancelling an in-progress registration

### PR DESCRIPTION
**Pull request by cintamani**
From: https://github.com/DEFRA/waste-carriers-acceptance-tests/issues/302

### CODE REVIEW

File: features/bo_new/dashboard/cancel.feature [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# TBD
+Feature: NCCC agent cancel in progress registration from back office
+  As an NCCC agent
+  I want to cancel an in-progress registration
+  So that we can correctly regulate the waste activity
+
```
Comment (cintamani) [inline features/bo_new/dashboard/cancel.feature:6]:
```suggestion
```

File: features/bo_new/dashboard/cancel.feature [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# TBD
```
Comment (andrewhick) [inline features/bo_new/dashboard/cancel.feature:1]: Suggested tags:
```suggestion
@bo_new @bo_dashboard @smoke
```

File: features/bo_new/dashboard/cancel.feature [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+Feature: NCCC agent cancel in progress registration from back office
+  As an NCCC agent
+  I want to cancel an in-progress registration
+  So that we can correctly regulate the waste activity
+
+Background:
+  Given I sign into the back office as "agency-refund-payment-user"
+
+Scenario: NCCC cancel a registration pending payment
```
Comment (andrewhick) [inline features/bo_new/dashboard/cancel.feature:10]: I suggest we say bank transfer here, to distinguish it from pending WorldPay:
```suggestion
Scenario: NCCC cancels a registration pending bank transfer
```

File: features/bo_new/dashboard/cancel.feature [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  Given a registration with outstanding balance and 0 copy cards has been submitted
+  When I cancel the registration
+  Then the registration has a status of "INACTIVE"
+
+Scenario: NCCC cancel a registration pending conviction checks
```
Comment (andrewhick) [inline features/bo_new/dashboard/cancel.feature:15]: Tiny grammar tweak:
```suggestion
Scenario: NCCC cancels a registration pending conviction checks
```

File: features/bo_new/dashboard/cancel.feature [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  Given a registration with outstanding convictions checks has been submitted
+  When I cancel the registration
+  Then the registration has a status of "INACTIVE"
+
+Scenario: NCCC select cancel and then decides to keep the registration
```
Comment (andrewhick) [inline features/bo_new/dashboard/cancel.feature:20]: Tiny grammar tweak:
```suggestion
Scenario: NCCC selects cancel and then decides to keep the registration
```

File: features/step_definitions/back_office/cancel_steps.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+When("I cancel the registration") do
+  visit_registration_details_page(@reg_number)
```
Comment (andrewhick) [inline features/step_definitions/back_office/cancel_steps.rb:2]: Should that be
```suggestion
  visit_registration_details_page(@pending_reg_number)
```
?

File: features/step_definitions/back_office/cancel_steps.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+  @bo.registration_details_page.cancel_link.click
+  @journey.standard_page.submit
+end
+
+When("I am about to cancel the registration and change my mind") do
+  visit_registration_details_page(@reg_number)
```
Comment (andrewhick) [inline features/step_definitions/back_office/cancel_steps.rb:9]: Same again?
```suggestion
  visit_registration_details_page(@pending_reg_number)
```

File: features/step_definitions/journey/registration_steps.rb [hunk 8]
```diff
@@ -222,6 +222,14 @@
   puts "Registration " + @reg_number + " seeded with #{copy_cards} copy cards and outstanding balance"
 end
 
+Given("a registration with outstanding convictions checks has been submitted") do
+  seed_data = SeedData.new("outstanding_convictions_checks_registration.json")
+  @reg_number = seed_data.reg_number
```
Comment (andrewhick) [inline features/step_definitions/journey/registration_steps.rb:227]: Should this be 
```suggestion
  @pending_reg_number = seed_data.reg_number
```
to be consistent with the rest of the feature?

File: features/step_definitions/journey/registration_steps.rb [hunk 9]
```diff
@@ -222,6 +222,14 @@
...
+  @seeded_data = seed_data.seeded_data
+
+  puts "Registration " + @reg_number + " seeded with outstanding conviction checks"
```
Comment (andrewhick) [inline features/step_definitions/journey/registration_steps.rb:230]:
```suggestion
  puts "Registration " + @pending_reg_number + " seeded with outstanding conviction checks"
```

File: features/step_definitions/back_office/cancel_steps.rb (hunk 6 above)
Comment (cintamani) [reply features/step_definitions/back_office/cancel_steps.rb:2]: No, the only step that sets pending_reg_number is the one created for the permissions checks. All others, still use the normal @reg_number

File: features/step_definitions/back_office/cancel_steps.rb (hunk 7 above)
Comment (cintamani) [reply features/step_definitions/back_office/cancel_steps.rb:9]: Same as https://github.com/DEFRA/waste-carriers-acceptance-tests/pull/304#discussion_r428074525

**Pull request merged**